openstack/imageservice/v2/imageimport/results.go

```diff
@@ -0,0 +1,32 @@
+package imageimport
+
+import "github.com/gophercloud/gophercloud"
+
+type commonResult struct {
+	gophercloud.Result
+}
+
+// GetResult represents the result of a get operation. Call its Extract method
+// to interpret it as ImportInfo.
+type GetResult struct {
+	commonResult
+}
+
+// ImportInfo represents information data for the Import API.
+type ImportInfo struct {
+	ImportMethods ImportMethods `json:"import-methods"`
+}
+
+// ImportMethods contains information about available Import API methods.
+type ImportMethods struct {
+	Description string   `json:"description"`
+	Type        string   `json:"type"`
+	Value       []string `json:"value"`
+}
+
+// Extract is a function that accepts a result and extracts ImportInfo.
+func (r commonResult) Extract() (*ImportInfo, error) {
+	var s *ImportInfo
+	err := r.ExtractInto(&s)
+	return s, err
+}
```

Comment:
@ozerovandrei If `ImportInfo` is only being used as a "shell" for `ImportMethods`, then we should be able to do this:

```go
func (r commonResult) Extract() (*ImportMethods, error) {
  var s struct {
    ImportMethods ImportMethods `json:"import-methods"`
  }
  err := r.ExtractInto(&s)
  return s.ImportMethods, err
}
```

This removes the need for the `ImportInfo` struct.

Reply:
I disagree because in the [API reference](https://developer.openstack.org/api-ref/image/v2/#import-methods-and-values-discovery) it says:
`In the Image API v2.6, this discovery call contains only the import-methods field.`

And if you research the [Importing API spec](https://specs.openstack.org/openstack/glance-specs/specs/mitaka/approved/image-import/image-import-refactor.html), you will find the much bigger schema for the `GET v2/info/import` value discovery (it's really huge).

So i think `ImportInfo` response will grow bigger in the future so it's better to have current two structures in that case.

Reply:
OK sounds good.